src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -129,7 +129,7 @@ private BoundExpression RewriteDeconstruction(BoundTupleExpression left, Convers
             for (int i = 0; i < leftTargets.Count; i++)
             {
                 BoundExpression resultPart;
-                if (leftTargets[i].HasNestedVariables)
+                if (leftTargets[i].NestedVariables is object)
```

Comment:
Could we annotate HasNestedVariables somehow to make NestedVariables null-checked?

Reply:
Not yet. We're going to be adding an API for this in C# 9, `MemberNotNull`, but there's nothing there for it today.